compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -842,38 +1043,32 @@ int32_t TR_OSRLiveRangeAnalysis::perform()
             {
             liveLocalIndexToSymRefNumberMap[liveLocalIndex] = nextSymRef->getReferenceNumber();
             nextSymRef->getSymbol()->setLocalIndex(0);
-            if (nextSymRef->getReferenceNumber() > maxSymRefNumber)
-               maxSymRefNumber = nextSymRef->getReferenceNumber();
-            if (comp()->getMethodSymbol()->sharesStackSlot(nextSymRef))
-               slotSharingVars->set(liveLocalIndex);
             }
          }
       }
 
-   for (i=0;i<numPendingSlots;i++)
+   if (containsPendingPushes)
       {
-      List<TR::SymbolReference> symRefsAtThisSlot = (*pendingPushSymRefs)[i];
+      for (i=0;i<numPendingSlots;i++)
+         {
+         List<TR::SymbolReference> symRefsAtThisSlot = (*pendingPushSymRefs)[i];
```

Comment:
Should this copy the list or be a reference?

Reply:
All of these should have been references, an error made prior to this change. They have all been updated to use references.